tools/not.cpp

```diff
@@ -63,14 +64,23 @@ int main(int argc, const char **argv) {
     result = -3;
 #endif
 
-  if (result < 0) {
+  // On POSIX systems, result is a composite value of the exit status and,
+  // potentially, the signal that caused termination of the command.
+  int retcode = WEXITSTATUS(result);
+  int signal = WTERMSIG(result);
+
```

Comment:
You probably don't have access to these function on Windows, what about an 
```cpp
# ifdef _Windows
//Windows stuff
#else
// Probably not the best way to handle POSIX
WEXITSTATUS
WTERMSIG
#endif
```

Reply:
> You probably don't have access to these functions on Windows, what about a structure similar to this:
> 
> ```c++
> #ifdef _Win32
> //Windows stuff
> #else
> // Probably not the best way to check POSIX
> WEXITSTATUS
> WTERMSIG
> #endif
> ```

For Windows, is reasonable to assume that if the return code < 0, then the command aborted? This seems to suggest that it is: [https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/system-wsystem?view=msvc-170](https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/system-wsystem?view=msvc-170).

Why do you have a comment saying, "Probably not the best way to check POSIX"?

Reply:
I am not an expert on Windows stuff but based on the doc, I think that it means if there is problem with command-interpreter (cmd in this case), then it will return -1 and set errno. If cmd is fine and the command has problem, the return value will be cmd's return value.
e.g. if you test it with `int val = std::system("InvalidCommand");` val will be set to 0. My best guess in this case is to set retcode to the return value of `std::system()`, and if `returnVal==-1` then you set signal to errno.

Just because it isn't Windows doesn't mean it is POSIX-compliant? You can `#if defined(WIFSTOPPED) && defined(WSTOPSIG)` and then get exitstatus and sig, if it isn't defined then just take whatever value from `std::system`. I guess you can just leave it as is, unless we need to support system that is not Windows nor POSIX. because each of them may have their own implementation and ask you to match all the cases would be impossible.

Reply:
That's a fair point. I have added checks for those macros, checked explicitly for `errno` on Windows and for the `WEXITSTATUS` and `WTERMSIG` macros. 

I can only think of Solaris as a non-POSIX system other than Windows that is likely to be of interest. It seems as if those macros [are defined](https://docs.oracle.com/cd/E36784_01/html/E36873/stdlib-3head.html) there. 

On any other system, I force a build error. I think it is safer than just using the result of `std::system` in such cases, but I can use a different strategy if folks feel differently.

Reply:
LGTM, thanks!